cap/modules/experiments/views/atlas.py

```diff
@@ -72,24 +92,10 @@ def yadage_workflow_submit():
 @atlas_bp.route('/glance/<id>', methods=['GET'])
 @atlas_permission.require(403)
 def get_glance_by_id(id):
-    """Retrieves GLANCE analysis data by given id."""
-    access_token = get_glance_token()
-
-    if not access_token:
-        abort(503, 'External server replied with an error.')
-
-    url = current_app.config.get('GLANCE_GET_BY_ID_URL').format(id=id)
-
-    try:
-        resp = requests.get(url=url, headers={
-            'Content-Type': 'application/json',
-            'Authorization': 'Bearer {}'.format(access_token)
-        })
-
-        data = resp.json()
-        item = data['items'][0] if data['items'] else {}
-
-    except (KeyError, ValueError):
-        abort(503, 'External server replied with an error.')
+    """Retrieves GLANCE analysis data by given id (route)."""
+    resp = _get_glance_by_id(id)
+    data = resp.json()
 
-    return jsonify(item)
+    # basically, if it has en error, send the error message
+    items = data['items'][0] if data['items'] else data
+    return jsonify(items)
```

Comment:
so in case of exception we jsonify, to then do resp.json() and jsonify again. maybe there's a better way ;)